docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+
+### 1.2.2. Preview
+
+In `PREVIEW` mode, our tool will then use the data provided and any available model information to show a "snapshot" of our data and how we will feed it to the model.
+This snapshot will include dynamically populated text classifier information like token translations for our classifier when available.
+
+### 1.2.3. TRAIN
+
+This mode will take the prepped data and call the model's training function.
+Additionally, the User may use the supplied fields to adjust model parameters, train using an existing model, or create an entirely new one.
+Finally, the tool will return a training results report to the UI and export a "production" version of the model.
+
+### 1.2.4. Predict
+
+In `PREDICT` mode, the tool will take text input via the input anchor, then run predictions on them and output those results to its output anchor.
+
+
+## Create a Workspace
+
+Run the following, responding as prompted.
+
+```powershell
+$ mkdir tensorflow-ui-examples
+$ cd ./tensorflow-ui-examples/
+tensorflow-ui-examples $ ayx_plugin_cli sdk-workspace-init
+Package Name: PythonSdkTensorflowExamples
+Tool Category [Python SDK Examples]: Tensorflow Python SDK Examples
+Description []: Tools using Tensorflow.
+Author []: Shawn Owen
+Company []: Alteryx
+Backend Language (python): python
+[Workspace initialization] started
+[Workspace initialization] .  Create configuration directory
+// output truncated //
+[Generating config files] started
+[Generating config files] .  generate_config_files:generate_config_xml
+[Generating config files] Generating top level config XML file...
+[Generating config files] finished
+```
+Once the above is complete, we recommend you create your Python environment with `venv` or other supported virtual environment modules.
+This venv will allow you to develop on your machine while deploying with "prod" requirements. We'll also briefly cover how to do that later in the guide.
+
+
+## 1.2.5. Create a Plugin
+Next, use the `create-ayx-plugin` command and reply to the prompts to generate our tool template code. For this tool, we want to use the `single-input-single-output`. (may switch to optional*)
+
+```powershell
+tensorflow-ui-examples $ ayx_plugin_cli create-ayx-plugin
+Tool Name: TextClassifier
+Tool Type (input, multiple-inputs, multiple-outputs, optional, output, single-input-single-output, multi-connection-input-anchor) [optional]:
+Description []: Create, train, deploy, and use a Tensorflow based text classifier!
+Tool Version [1.0]:
+DCM Namespace []:
+Creating single-input-single-output plugin: Filter UI Tool
+[Create plugin] started
+[Create plugin] .  Create plugin
+[Create plugin] Installing UI components via npm
+[Create plugin] Creating Alteryx Plugin...
+// output truncated //
+[Generating test files for FilterUITool] finished
+```
+
+After the command finishes, you should have template files similar to below at the named locations for your UI and python SDK.
```

Comment:
TW review:

After the command finishes, you should have template files similar to those below at the named locations for your UI SDK and Python SDK.